### PR TITLE
Metal: Prevent usage of macOS-only APIs on non-macOS platforms

### PR DESCRIPTION
**Pull request by OatmealDome**
In the spirit of keeping the differences between my iOS fork and upstream minimal (and since the Metal backend already has support for iOS), here's a bunch of changes to make sure Metal APIs that are unavailable on non-macOS platforms are not used when targeting non-macOS.

### CODE REVIEW

**Comment by OatmealDome** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/VideoBackends/Metal/MTLMain.mm [hunk 1]
```diff
@@ -156,11 +159,11 @@ static bool WindowSystemTypeSupportsMetal(WindowSystemType type)
 
 void Metal::VideoBackend::PrepareWindow(WindowSystemInfo& wsi)
 {
-  if (wsi.type != WindowSystemType::MacOS)
-    return;
```
Comment (tellowkrinkle) [inline Source/Core/VideoBackends/Metal/MTLMain.mm:160]: Can you leave this in?  I don't really expect to receive any other window types, but just in case

**Review by tellowkrinkle — APPROVED** [review]

**Comment by JosJuice** [comment]
I don't know Apple, but this has been approved by the Apple knowers, so...

**Comment by ShizCalev** [comment]
I'm getting Invalid Memory Access Locations errors on launch since this commit. 

Relevant issue report: https://bugs.dolphin-emu.org/issues/12995

**Comment by OatmealDome** [comment]
Are you sure this bisect is correct? This PR changes code that only runs on macOS, and you are running Windows 10.

**Comment by ShizCalev** [comment]
Going through all the recent builds one by one, 5.0-16986 (this PR) is the first build where the issue starts to occur. 5.0-16979 and prior doesn't have the issue.

image

**Comment by OatmealDome** [comment]
Do you have ESET NOD32?

**Comment by ShizCalev** [comment]
Hah, yes I nod 32.

**Comment by OatmealDome** [comment]
Please disable NOD32 or whitelist Dolphin. For whatever reason, NOD32 thinks that our fastmem code is something suspicious and blocks it. (If you are using other emulators with fastmem, this may also apply to them.)

**Comment by ShizCalev** [comment]
My issue was fixed in [5.0-17025](https://dolphin-emu.org/download/dev/b6f5499c7cc7763b622ca4322192e6c83da706f1/) via #10900. Thanks for the recommendation with ESET though!

**Pull request merged**